### ISSUE TITLE
Helm delete with purge flag doesn't clean release history

### ISSUE THREAD

**Issue by gabrtv** [kubernetes/helm#1403]
Helm delete with `--purge` flag doesn't clean release history.  This seems to confuse Tiller's state and causes problems when trying to deploy a new release.  Reproduce steps:

```$
$ helm install ./croc-hunter --name croc-hunter --namespace croc-hunter
$ helm delete croc-hunter --purge
$ helm install ./croc-hunter --name croc-hunter --namespace croc-hunter
Error: a release named "croc-hunter" already exists
$ kubectl get configmap -n kube-system
NAME             DATA      AGE
croc-hunter.v1   1         3h
$ helm delete croc-hunter --purge
Error: release: not found
```

**Comment by rimusz** [comment]
yes, i'm seeing the same issues

**Comment by technosophos** [comment]
Can you paste in `helm version`? Was this the Alpha.5 build?

**Comment by gabrtv** [comment]
```$
$ helm version
Client: &version.Version{SemVer:"v2.0.0-alpha.5", GitCommit:"a324146945c01a1e2dd7eaf23caf0e55fabfd3d2", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.0.0-alpha.5", GitCommit:"4de5c4aa578ed9ca173aff472f00d2167ec59a88", GitTreeState:"dirty"}
```

**Comment by technosophos** [comment]
I think that if you do a `helm init --canary` that problem should go away (you have to `kubectl delete deployment tiller-deployment` first, though). But I'm going to make sure this was fixed before closing the issue.

**Comment by fibonacci1729** [comment]
closed by: #1405
